lib/src/session.dart

```diff
@@ -0,0 +1,190 @@
+import "dart:async";
+import "dart:typed_data";
+
+import "package:wamp/src/helpers.dart";
+import "package:wamp/src/types.dart";
+import "package:wampproto/idgen.dart";
+import "package:wampproto/messages.dart";
+import "package:wampproto/session.dart";
+
+class Session {
+  Session(this._baseSession) {
+    _wampSession = WAMPSession(serializer: _baseSession.serializer);
+    Future.microtask(() async {
+      while (true) {
+        await _baseSession.receive().then(
+              (value) => _processIncomingMessage(_wampSession.receive(Uint8List.fromList((value as String).codeUnits))),
+            );
+      }
+    });
+  }
+
+  final BaseSession _baseSession;
+  late WAMPSession _wampSession;
+
+  final SessionScopeIDGenerator _idGen = SessionScopeIDGenerator();
+
+  int get _nextID => _idGen.next();
+
+  final Map<int, Completer<Result>> _callRequests = {};
+  final Map<int, RegisterRequest> _registerRequests = {};
+  final Map<int, Yield Function(Invocation)> _registrations = {};
```

Comment:
As far as user api is concerned this should have a new class Result AND Invocation defined in types. So instead of Yield we will use Result. NOTE Result wont be a wamp message type but rather a normal class. See wamp.py